### ISSUE TITLE
Getting FALSE DesiredState due to [SPProjectServerLicense] Resource

### ISSUE THREAD

**Issue by viguerre** [PowerShell/SharePointDsc#828]
When running 'Test-DscConfiguration -ComputerName <servername> -Detailed' on my DSC configuration, I get 'InDesiredState' as FALSE.  The resource that comes back as FALSE is the following:

[SPProjectServerLicense]PWALicense

I am not sure why this is happening.  I double checked the license key and it is the correct key.  Also, my DSC successfully enables and configures Project Server 2016.  Any help is greatly appreciated.  Thank you.
        
OS:  Windows Server 2016
Target Node:  Application Server
Version:  SharePoint Server 2016 Enterprise
Version of DSC module:  SharePointDSC 2.2.0.0

**Comment by viguerre** [comment]
Resource used:
`SPProjectServerLicense PWALicense
{
 Ensure                           = "Present"
 ProductKey                   = "XXXXX-XXXXX-XXXXX-XXXXX-XXXXX"
 PsDscRunAsCredential = $spsetup
 DependsOn                  = "[SPServiceAppPool]ProjectServerServiceAppPool"
 }`

**Comment by ykuijs** [comment]
Hi @viguerre I don't have an environment which has Project Server on it, so can you do me a favor and please share me the output of the Get-ProjectServerLicense cmdlet on your environment?

**Comment by ykuijs** [comment]
And what is the output of the following command:
```
Invoke-DscResource -Name SPProjectLicense -Method Get -Property @{Ensure = "Present"
                                                                  ProductKey = "XXXXX-XXXXX-XXXXX-XXXXX-XXXXX"
                                                                  PsDscRunAsCredential = $spsetup}
```

**Comment by ykuijs** [comment]
Of course make sure you first set $spsetup ;-)

**Comment by viguerre** [comment]
Good afternoon @ykuijs 

This is what I get when I run the Get-ProjectServerLicense in my environment:
```
Get-ProjectServerLicense
Project Server 2016 : Active
Activated: 6/18/2018
```
And when I add the above command, I get the following:

```
cmdlet Invoke-DscResource at command pipeline position 1
Supply values for the following parameters:
ModuleName: SPProjectServerLicense
Invoke-DscResource : Resource SPProjectLicense was not found.
At line:1 char:1
+ Invoke-DscResource -Name SPProjectLicense -Method Get -Property @{Ens ...
+ ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : NotSpecified: (:) [Invoke-DscResource], ArgumentException
    + FullyQualifiedErrorId : System.ArgumentException,Microsoft.PowerShell.DesiredStateConfiguration.Commands.InvokeDscResourceMethodCommand
```

**Comment by ykuijs** [comment]
Did you run that second command on your SP/Project server??

**Comment by viguerre** [comment]
yes I did

**Comment by ykuijs** [comment]
Woeps, my mistake. You have to specify SPProject**Server**License:
```
Invoke-DscResource -Name SPProjectServerLicense -Method Get -Property @{Ensure = "Present"
                                                                  ProductKey = "XXXXX-XXXXX-XXXXX-XXXXX-XXXXX"
                                                                  PsDscRunAsCredential = $spsetup}
```

**Comment by viguerre** [comment]
```
ConfigurationName    : 
DependsOn            : 
ModuleName           : SharePointDSC
ModuleVersion        : 2.2.0.0
PsDscRunAsCredential : 
ResourceId           : 
SourceInfo           : 
Ensure               : Absent
InstallAccount       : 
ProductKey           : XXXXX-XXXXX-XXXXX-XXXXX-XXXXX
PSComputerName       : localhost
```

**Comment by viguerre** [comment]
Interesting, its showing Ensure as 'Absent' when I set it at 'Present' in my configuration.

**Comment by ykuijs** [comment]
That means that the code doesn't detect the license properly

**Comment by ykuijs** [comment]
Let's try this:
```
$currentLicense = Get-ProjectServerLicense

if ($currentLicense -match "Project Server [0-9]{4} : (?<Status>[a-zA-Z]+)")
{
    if ($Matches.Status -eq "Active")
    {
        $status = "Present"
    }
    else
    {
        $status = "Absent"
    }

    return @{
        Ensure = $status
        ProductKey = $params.ProductKey
        InstallAccount = $params.InstallAccount
    }
}
else
{
    Write-Output -Message "Unable to determine the license status for Project Server"
    return @{
        Ensure = "Absent"
        ProductKey = $params.ProductKey
        InstallAccount = $params.InstallAccount
    }
}
```

**Comment by ykuijs** [comment]
The regular expression does not process the output of Get-ProjectServerLicense properly. I might have found the solution. Shared this with Victor. If he confirms the fix, I will add it to my Bugfix PR.

**Comment by ykuijs** [comment]
Closed via PR #826

**Comment by viguerre** [comment]
what module version will this fix be available in? 2.5.0.0?  I am still seeing the error in 2.3.0.0.  Thanks!

**Comment by viguerre** [comment]
oops, never mind, figured it out.
